src/astro/files/locations/__init__.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict, Optional, Type, Union
+from urllib.parse import urlparse
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.files.locations.gcs import GS
+from astro.files.locations.http import Http
+from astro.files.locations.local import Local
+from astro.files.locations.s3 import S3
+
+
+def location_factory(path: str, conn_id: Optional[str] = None) -> Location:
```

Comment:
I'm not so sure, don't you think this takes away the readability/simplicity of the code?

Reply:
I think it is not a matter of readability only, but to have a centralized place where we're importing all supported modules - without the need for that. How will we handle optional dependencies? Using the MissingPackage approach? 

We can still have the other approach having an explicit map to dot paths of modules.

Reply:
I think there are three approaches:

1. Have a central module that imports all the dependencies and uses the `MissingPackage` approach to handle optional dependencies.

2. Maintain an explicit dictionary mapping `types` to dot string module paths, without actually importing them. The modules would only be loaded if and when the user tried to use them, using `importlib` (https://github.com/astro-projects/astro/pull/320/commits/f8b220740a22d6b1bd117e0ee73c73bae35aa435#diff-a1f816ae9cccff9de86f230ac28a5e34c815fe6f51b99c7c66404e60375a1525R7-R12)
* Pro: simplifies the code, since we no longer need to worry about Missing Packages - while still having an explicit dictionary that is easy to maintain

3. Have an automatically generated mapping of `types` to modules dot string paths, for the default case (e.g. when a type & the implementation module have the same name), only adding explicit values when we want to match non-default types matching import paths.
* Pro: when we add a new database, we could in theory only care about adding a new database module - without having to remember to update any dictionary

Reply:
Had a conversation with @kaxil, will implement this approach.

Reply:
Done.